utils/incentive.js

```diff
@@ -1,86 +1,156 @@
+const { setHeaders, getResponseJSON, logIPAdddress, APIAuthorization, isParentEntity, isDateTimeFormat } = require('./shared');
 
 const incentiveCompleted = async (req, res) => {
     logIPAdddress(req);
     setHeaders(res);
 
     if(req.method === 'OPTIONS') return res.status(200).json({code: 200});
+    if(req.method !== 'POST') return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
 
-    if(req.method !== 'POST') {
-        return res.status(405).json(getResponseJSON('Only POST requests are accepted!', 405));
-    }
+    const authorized = await APIAuthorization(req);
 
-    const { APIAuthorization } = require('./shared');
-    const authorized = await APIAuthorization(req, true);
-    if(authorized instanceof Error){
-        return res.status(500).json(getResponseJSON(authorized.message, 500));
-    }
-    if(!authorized){
-        return res.status(401).json(getResponseJSON('Authorization failed!', 401));
-    }
+    if(authorized instanceof Error) return res.status(500).json(getResponseJSON(authorized.message, 500));
+    if(!authorized) return res.status(401).json(getResponseJSON('Authorization failed!', 401));
+
+    console.log(`Incentives API: Incentive Completed, accessed by: ${authorized.saEmail}`);
     
-    const { isParentEntity } = require('./shared');
     const {isParent, siteCodes} = await isParentEntity(authorized);
 
+    let responseArray = [];
+    let error = false;
+
     const data = req.body.data;
+
     for(let i = 0; i < data.length; i++) {
-        if(data[i].token && data[i].round && data[i].caseNumber) {
-            const token = data[i].token;
-            const round = data[i].round;
-            if(round !== 'baseline' && round !== 'followup1' && round !== 'followup2' && round !== 'followup3') continue;
-            const caseNumber = data[i].caseNumber;
-            
-            const { incentiveConcepts } = require('./shared');
-            const roundConcept = incentiveConcepts[round];
-            let obj = {};
-            if(data[i].incentiveIssued) {
-                obj[`${roundConcept}.${incentiveConcepts['caseNumber']}`] = caseNumber;
-                if(data[i].incentiveChosen) obj[`${roundConcept}.${incentiveConcepts['incentiveChosen']}`] = data[i].incentiveChosen;
-                obj[`${roundConcept}.${incentiveConcepts['incentiveIssued']}`] = data[i].incentiveIssued ? 353358909 : 104430631;
-                obj[`${roundConcept}.${incentiveConcepts['incentiveIssuedAt']}`] = new Date().toISOString();
+
+        const { token, round, incentiveIssued, incentiveIssuedAt, incentiveRefused, incentiveRefusedAt, incentiveChosen, caseNumber } = data[i];
+        
+        if(token === undefined) {
+            error = true;
+            responseArray.push({'Invalid Request': {'Token': 'UNDEFINED', 'Errors': 'Token not defined in data object.'}});
+            continue;
+        }
+
+        const { getParticipantData } = require('./firestore');
```

Comment:
Better import functions outside the `for` loop, so that they're not repeatedly imported.